src/main/webapp/scripts/viewCard.js

```diff
@@ -0,0 +1,127 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the 'License');
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an 'AS IS' BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const MAP_THEME = [
+  {
+    featureType: 'poi.business',
+    stylers: [
+      {
+        visibility: 'off'
+      }
+    ]
+  },
+  {
+    featureType: 'poi.park',
+    elementType: 'labels.text',
+    stylers: [
+      {
+        visibility: 'off'
+      }
+    ]
+  }
+];
+
+/**
+ * Map declaration and initial setup
+ * At the beginning, retrieves the API_KEY from an external file
+ */
+const initMap = () => {
+  jQuery.get('../../../../API_KEY.txt', function (textString) {
+    const API_KEY = textString;
+
+    // Create the script tag, set the appropriate attributes.
+    const scriptMapTag = document.createElement('script');
+    scriptMapTag.src = `https://maps.googleapis.com/maps/api/js?key=${API_KEY}&callback=drawMap`;
+    scriptMapTag.defer = true;
+
+    document.head.appendChild(scriptMapTag);
+  });
+};
+
+/**
+ * Query salecard from servlet
+ * TODO: fetch from servlet
+ */
+const querySalecard = () => {
```

Comment:
Same here, looks very similar to the method that's already been defined above.

Reply:
I added a `map-commons.js` file that removes duplicated code of the map, but I am planning to remove `querySalecard` in the integration with angel's new servlets. What do you think about that ?